src/store/actions/Chat.ts

```diff
@@ -0,0 +1,161 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import firebase from "firebase/app";
+
+export const SEND_GLOBAL_CHAT: string = "SEND_GLOBAL_CHAT";
+export const SEND_ROOM_CHAT: string = "SEND_ROOM_CHAT";
+export const SEND_PRIVATE_CHAT: string = "SEND_PRIVATE_CHAT";
+export const SEND_TABLE_CHAT: string = "SEND_TABLE_CHAT";
+
+interface BaseSendChatFields {
+  venueId: string;
+  text: string;
+  from: string;
+}
+interface BaseSendChatToFields extends BaseSendChatFields {
+  to: string;
+}
+
+interface SendGlobalChatFields extends BaseSendChatFields {}
+interface SendPrivateChatFields extends Omit<BaseSendChatToFields, "venueId"> {}
+interface SendRoomChatFields extends BaseSendChatToFields {}
+interface SendTableChatFields extends BaseSendChatToFields {}
+
+interface SendGlobalChatAction extends SendGlobalChatFields {
+  type: typeof SEND_GLOBAL_CHAT;
```

Comment:
I don't think that works as intended. Not sure about the goals ofc, but this could work:

```
interface SendPrivateChatAction extends SendPrivateChatFields {
  type: ChatMessageType.Global;
}
```

And we can delete those consts(SEND_ROOM_CHAT) and export `ChatMessages` enum instead

Reply:
Hm, this is an unfortunate one. I'd like to do this simplification but unfortunately the `type` values (`global`, `room`, `table`, `private`) are already in the database. Doing that change would make it impossible to view old chats, because our chats would save to `SEND_{GLOBAL,PRIVATE,TABLE,ROOM}_CHAT`.

So it looks to me like we have to leave it this way unfortunately.

Good catch though, glad you're spotting these!

Reply:
typeof SEND_GLOBAL_CHAT results into string, which is not as good as it could be. A good workaround this could be either creating 
enum SendChatTypes = {
    
}

or create 
type SendTableChat = "SEND_TABLE_CHAT";
export const SEND_TABLE_CHAT: SendTableChat = "SEND_TABLE_CHAT";